[ItsTYtan] iP

ItsTYtan opened the pull request:
# TY's slave
> Withi**NUS** beyond ***SELF*** - NUS ([source](https://www.instagram.com/teamnus/))

TY's slave helps you to remember stuff you needa do, It's

- text-based
- hard to learn
- embarrassing to use

All you need to do is,

1. download it from [here](https://9animetv.to/watch/redo-of-healer-15754?ep=51259)
2. double click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Helping you invest with an annual compounded return of 35% (coming never)

Here is what I have done so far:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

ItsTYtan committed:
Rename, Greet, Exit

ItsTYtan committed:
Echo

ItsTYtan committed:
Add, List

ItsTYtan committed:
Mark as Done

ItsTYtan committed:
ToDos, Events, Deadlines

ItsTYtan committed:
Automated Text UI Testing

ItsTYtan committed:
Handle Errors

ItsTYtan committed:
Delete

ItsTYtan committed:
Use Enums

ItsTYtan committed:
Implement a save function

ItsTYtan committed:
Merge branch 'branch-Level-7'

ItsTYtan committed:
Add date and time functionality

ItsTYtan committed:
Merge branch 'branch-Level-8'

ItsTYtan committed:
Add more OOP into the code

ItsTYtan committed:
Merge branch 'branch-A-MoreOOP'

ItsTYtan committed:
Merge remote-tracking branch 'origin/add-gradle-support'

ItsTYtan committed:
Add JUnit Tests

ItsTYtan committed:
Add JavaDoc comments

ItsTYtan committed:
Merge branch 'branch-A-JavaDoc'

ItsTYtan committed:
Tweak the code to comply with a coding standard

ItsTYtan committed:
Merge branch 'branch-A-CodingStandard'

ItsTYtan committed:
Enable users to find a task

ItsTYtan committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/commands/FindCommand.java
#	src/main/java/duke/tasks/Task.java

ItsTYtan committed:
Use CheckStyle

ItsTYtan committed:
Merge branch 'branch-A-Checkstyle'

ItsTYtan committed:
Add a GUI

ItsTYtan committed:
Merge branch 'branch-Level-10'

prawnzyy commented on src/main/java/duke/commands/FindCommand.java:
```diff
@@ -0,0 +1,34 @@
+            }
+            return output;
+        } catch (InvalidTaskIndexException ex) {
+            return "Something went wrong.";
```
Maybe add something more descriptive so as to let users know the error in their input.

nicrandomlee replied:
Yup agreed! Perhaps you can also add in the correct format that the user is suppose to enter so that they have a point of reference 😄😄

prawnzyy commented on src/main/java/duke/tasks/Event.java:
```diff
@@ -0,0 +1,68 @@
+            throw new IncompleteDescriptionException();
+        }
+        String taskName = tmp[0];
+        String[] tmp2 = tmp[1].split(" /to ");
```
Maybe avoid making names such as tmp2 and use names like timings or dates?

prawnzyy reviewed (commented):
Overall, the code was well split and all the classes and methods were well defined. If possible, try to make the try catch blocks return a more descriptive statement to the user if they make an input error. Other than that, no issues. LGTM!

nicrandomlee commented on src/main/java/duke/tasks/Deadline.java:
```diff
@@ -0,0 +1,58 @@
+     */
+    public static Deadline create(String taskDesc) throws IncompleteDescriptionException {
+        String[] tmp = taskDesc.split(" /by ");
+        if (tmp.length <= 1) {
```
Perhaps it would be clearer to separate logical units within a block with one blank line!

nicrandomlee commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,107 @@
+        try {
+            CommandHeaders cmdHdr = CommandHeaders.valueOf(cmdHdrString.toUpperCase());
+            switch (cmdHdr) {
+            case BYE:
```
Good job on correct no-indentation for case clauses :)

nicrandomlee reviewed (commented):
Overall, I found your code really easy to read and most of my comments are just nitpicking on small details! I also like that you have OOP-ified your code to separate commands, tasks and exceptions and implemented relevant test cases. Good job and thank you for allowing me to learn from you 👍

ItsTYtan committed:
Add assertions

ItsTYtan committed:
Improve code quality

ItsTYtan committed:
Merge pull request #1 from ItsTYtan/branch-A-Assertions

Add assertions

ItsTYtan committed:
Merge pull request #2 from ItsTYtan/branch-A-CodeQuality

Improve code quality

ItsTYtan committed:
Add more flexibility in search e.g., find items even if the keyword matches the item only partially.

ItsTYtan committed:
Merge branch 'branch-BCD-Extension'

ItsTYtan committed:
Add user guide, Ui.png, and finalise features

ItsTYtan committed:
final code fixes

ItsTYtan committed:
Tweak the code to be suitable for release

ItsTYtan committed:
(Dummy commit for A-FullCommitMessage on the A-Assertions commit I
forgot to do earlier)

Add assertions to various points in the code

The program does not have features to indicate a possible
bug in the code during runtime.

A bug during runtime should be addressed as soon as possible so
that the damage is minimized.

Let's add assertions to document important assumptions that
should hold at various points in the code.

Assertion failures indicates a possible bug and allows us to take
drastic action such as termination of the program.

ItsTYtan committed:
(Dummy commit for A-FullCommitMessage on the A-CodeQuality commit I
forgot to do earlier)

Improve code quality

The code is hard to read and not very extensible.

Code may need to be read, understood, and modified by other
developers later on.

Let's clean up the code, mainly by breaking down long methods into
shorter ones, reducing the levels of abstraction within code
fragments.

This is all done to improve readability.
